polygeist/tools/cgeist/Test/Verification/globals.c

```diff
@@ -11,22 +12,43 @@ void kernel_deriche() {
     run(&local, &local_init, &internal, &internal_init, &external);
 }
 
-// CHECK-DAG:   memref.global @external : memref<1xi32>
-// CHECK-DAG:   memref.global "private" @internal_init : memref<1xi32> = dense<5>
-// CHECK-DAG:   memref.global "private" @internal : memref<1xi32> = uninitialized
-// CHECK-DAG:   memref.global @local_init : memref<1xi32> = dense<4>
-// CHECK-DAG:   memref.global @local : memref<1xi32> = uninitialized
+// LLVM-DAG: @local = global i32 undef
+// LLVM-DAG: @local_init = global i32 4
+// LLVM-DAG: @internal = private global i32 undef
+// LLVM-DAG: @internal_init = private global i32 5
+// LLVM-DAG: @external = external global i32
+// LLVM-LABAL: define void @kernel_deriche() {
+// LLVM:        call void @run(i32* @local, i32* @local_init, i32* @internal, i32* @internal_init, i32* @external)
+// LLVM-NEXT:   ret void
+// LLVM-NEXT: }
+// LLVM: declare void @run(i32*, i32*, i32*, i32*, i32*)
+
+// CHECK-DAG:   memref.global @external : memref<i32>
+// CHECK-DAG:   memref.global "private" @internal_init : memref<i32> = dense<5>
+// CHECK-DAG:   memref.global "private" @internal : memref<i32> = uninitialized
+// CHECK-DAG:   memref.global @local_init : memref<i32> = dense<4>
+// CHECK-DAG:   memref.global @local : memref<i32> = uninitialized
 // CHECK:   func @kernel_deriche()
-// CHECK-NEXT:     %0 = memref.get_global @local : memref<1xi32>
-// CHECK-NEXT:     %cast = memref.cast %0 : memref<1xi32> to memref<?xi32>
-// CHECK-NEXT:     %1 = memref.get_global @local_init : memref<1xi32>
-// CHECK-NEXT:     %cast_0 = memref.cast %1 : memref<1xi32> to memref<?xi32>
-// CHECK-NEXT:     %2 = memref.get_global @internal : memref<1xi32>
-// CHECK-NEXT:     %cast_1 = memref.cast %2 : memref<1xi32> to memref<?xi32>
-// CHECK-NEXT:     %3 = memref.get_global @internal_init : memref<1xi32>
-// CHECK-NEXT:     %cast_2 = memref.cast %3 : memref<1xi32> to memref<?xi32>
-// CHECK-NEXT:     %4 = memref.get_global @external : memref<1xi32>
-// CHECK-NEXT:     %cast_3 = memref.cast %4 : memref<1xi32> to memref<?xi32>
-// CHECK-NEXT:     call @run(%cast, %cast_0, %cast_1, %cast_2, %cast_3) : (memref<?xi32>, memref<?xi32>, memref<?xi32>, memref<?xi32>, memref<?xi32>) -> ()
+// CHECK-NEXT:     %0 = memref.get_global @local : memref<i32>
+// CHECK-NEXT:     %alloca = memref.alloca() : memref<1xindex>
```

Comment:
%alloca -> %shape ?

Reply:
Why?

Reply:
Looks like this feature is not supported: https://github.com/tensorflow/mlir/issues/185